packages/cw721/README.md

```diff
@@ -0,0 +1,120 @@
+# CW721 Spec: Non Fungible Tokens
+
+CW721 is a specification for non-fungible tokens based on CosmWasm.
+The name and design is based on Ethereum's ERC721 standard,
+with some enhancements. The types in here can be imported by 
+contracts that wish to implement this  spec, or by contracts that call 
+to any standard cw721 contract.
+
+The specification is split into multiple sections, a contract may only
+implement some of this functionality, but must implement the base.
+
+## Base
+
+This handles ownership, transfers, and allowances. These must be supported
+as is by all CW721 contracts. Note that all tokens must have an owner, 
+as well as an ID. The ID is an arbitrary string, unique within the contract.
+
+### Messages
+
+`TransferNft{recipient, token_id}` - 
+This transfers ownership of the token to `recipient` account. This is 
+designed to send to an address controlled by a private key and *does not* 
+trigger any actions on the recipient if it is a contract.
+
+Requires `token_id` to point to a valid token, and `env.sender` to be 
+the owner of it, or have an allowance to transfer it. 
+
+`SendNft{contract, token_id, msg}` - 
+This transfers ownership of the token to `contract` account. `contract` 
+must be an address controlled by a smart contract, which implements
+the CW721Receiver interface. The `msg` will be passed to the recipient 
+contract, along with the token_id.
+
+Requires `token_id` to point to a valid token, and `env.sender` to be 
+the owner of it, or have an allowance to transfer it. 
+
+`Approve{spender, token_id, expires}` - Grants permission to `spender` to
+transfer or send the given token. This can only be performed when
+`env.sender` is the owner of the given `token_id` or an `operator`. 
+There can multiple spender accounts per token, and they are cleared once
+the token is transfered or sent.
+
+`Revoke{spender, token_id}` - This revokes a previously granted permission
+to transfer the given `token_id`. This can only be granted when
+`env.sender` is the owner of the given `token_id` or an `operator`.
+
+`ApproveAll{operator, expires}` - Grant `operator` permission to transfer or send
+all tokens owned by `env.sender`. This approval is tied to the owner, not the
+tokens and applies to any future token that the owner receives as well.
+
+`RevokeAll{operator}` - Revoke a previous `ApproveAll` permission granted
+to the given `operator`.
+
+### Queries
+
+`OwnerOf{token_id}` - Returns the owner of the given token,
+as well as anyone with approval on this particular token.
+If the token is unknown, returns an error. Return type is
+`OwnerResponse{owner}`.
+
```

Comment:
Should there also be a query for all accounts approved to spend a specific NFT?

Reply:
This is inside `OwnerResponse` type and explained in the lines above